core/store/src/trie/shard_tries.rs

```diff
@@ -285,13 +319,20 @@ impl ShardTries {
                 .last()
                 .expect("Committed entry should have at least one change")
                 .data;
+            let flat_state_delta_key = KeyForStateChanges::from_raw_key(block_hash, &key);
             match last_change {
                 Some(value) => {
                     let value_ref_ser = near_primitives::state::ValueRef::create_serialized(value);
-                    store_update.set(DBCol::FlatState, &key, &value_ref_ser)
+                    store_update.set(
+                        DBCol::FlatStateDeltas,
+                        flat_state_delta_key.as_ref(),
```

Comment:
I wonder if we should store FlatStateDeltas as (block_hash, FlatStateDelta), instead of storing the individual key value pairs. It would be easier that way to load the delta from disk to memory when start up, which is mostly what storing delta is used for.

Reply:
Good point